src/components/FilterSearch.tsx

```diff
@@ -99,56 +108,55 @@ export function FilterSearch({
   );
 
   useEffect(() => {
-    if (currentFilter && fieldValueFilters?.find(f =>
-      isDuplicateFieldValueFilter(f, currentFilter) && !f.selected
+    if (currentFilter && staticFilters?.find(f =>
+      isDuplicateStaticFilter(f.filter, currentFilter) && !f.selected
     )) {
       clearFilterSearchResponse();
       setCurrentFilter(undefined);
       setFilterQuery('');
     }
-  }, [clearFilterSearchResponse, currentFilter, fieldValueFilters]);
+  }, [clearFilterSearchResponse, currentFilter, staticFilters]);
 
   const sections = useMemo(() => {
     return filterSearchResponse?.sections.filter(section => section.results.length > 0) ?? [];
   }, [filterSearchResponse?.sections]);
 
   const hasResults = sections.flatMap(s => s.results).length > 0;
 
-  const handleDropdownEvent = useCallback((value, itemData, select) => {
+  const handleSelectDropdown = useCallback((_value, _index, itemData) => {
     const newFilter = itemData?.filter as FieldValueStaticFilter;
     const newDisplayName = itemData?.displayName as string;
-    if (newFilter && newDisplayName) {
-      if (select) {
-        if (currentFilter) {
-          searchActions.setFilterOption({ filter: currentFilter, selected: false });
-        }
-        searchActions.setFilterOption({ filter: newFilter, displayName: newDisplayName, selected: true
-        });
-        setCurrentFilter(newFilter);
-        setFilterQuery(newDisplayName);
-        executeFilterSearch(newDisplayName);
-        if (searchOnSelect) {
-          searchActions.setOffset(0);
-          searchActions.resetFacets();
-          executeSearch(searchActions);
-        }
-      } else {
-        setFilterQuery(value);
-        executeFilterSearch(value);
+    if (!newFilter || !newDisplayName) {
+      return;
+    }
+
+    if (onSelect) {
+      if (searchOnSelect) {
+        console.warn('Both searchOnSelect and onSelect props were passed to the component.'
+        + ' Using onSelect instead of searchOnSelect as the latter is deprecated.');
```

Comment:
Is "using" the write tense?

Reply:
I was trying to indicate that both props were encountered, so the component is **using** `onSelect` instead of `searchOnSelect` in the present moment. so I think it is correct?

Reply:
Ah sorry I misread